### PR TITLE
Fix output property missing for airflow version < 2.4.0

### PR DESCRIPTION
**Pull request by sunank200**
# Description
## What is the current behavior?

The output was implemented in 2.4.0 according to the release notes (see [here](https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html#id13))

Add output property to MappedOperator https://github.com/apache/airflow/pull/25604

closes: #1359 

## What is the new behavior?

- Catch exception for airflow version < 2.4.0 and use `XComArg(...)` instead.

## Does this introduce a breaking change?
No

### Checklist
- [x] Created tests which fail without the change (if possible)
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: python-sdk/src/astro/sql/operators/cleanup.py [hunk 1]
```diff
@@ -224,17 +224,38 @@ def resolve_tables_from_tasks(  # noqa: C901
         for task in tasks:
             try:
                 if isinstance(task, OPERATOR_CLASSES_WITH_TABLE_OUTPUT):
-                    t = task.output.resolve(context)
-                    if isinstance(t, BaseTable):
-                        res.append(t)
+                    try:
+                        # works on airflow version >= 2.4.0
```
Comment (feluelle) [inline python-sdk/src/astro/sql/operators/cleanup.py:228]: If we now the exact airflow version, can't we decide based on this which path to go?

File: python-sdk/src/astro/sql/operators/cleanup.py [hunk 2]
```diff
@@ -224,17 +224,38 @@ def resolve_tables_from_tasks(  # noqa: C901
...
+                        if isinstance(t, BaseTable):
+                            res.append(t)
+                    except AttributeError:
+                        # works on airflow version < 2.4.0
+                        from airflow.models.xcom_arg import XComArg
+
+                        task_output = XComArg(operator=self)
+                        for t in task_output.resolve(context):
+                            if isinstance(t, BaseTable):
+                                res.append(t)
```
[the remark this answers is not in the document]
Comment (sunank200) [reply python-sdk/src/astro/sql/operators/cleanup.py]: [Remove the for loop](https://github.com/astronomer/astro-sdk/pull/1385/commits/1bea75404a4ca407be56deee72a2bdd3a0bef8f2)

File: python-sdk/src/astro/sql/operators/cleanup.py (hunk 1 above)
Comment (sunank200) [reply python-sdk/src/astro/sql/operators/cleanup.py]: [Done](https://github.com/astronomer/astro-sdk/pull/1385/commits/230add9e1d087f9fa6b1ae007e684aa8360db1a6)

**Review by feluelle — APPROVED** [review]

File: python-sdk/src/astro/sql/operators/cleanup.py [hunk 3]
```diff
@@ -224,17 +224,38 @@ def resolve_tables_from_tasks(  # noqa: C901
...
                 if isinstance(task, OPERATOR_CLASSES_WITH_TABLE_OUTPUT):
-                    t = task.output.resolve(context)
-                    if isinstance(t, BaseTable):
-                        res.append(t)
+                    try:
+                        # works on airflow version >= 2.4.0
+                        t = task.output.resolve(context)
+                        if isinstance(t, BaseTable):
+                            res.append(t)
+                    except AttributeError:
```
[the remark this answers is not in the document]
Comment (sunank200) [reply python-sdk/src/astro/sql/operators/cleanup.py]: [Done](https://github.com/astronomer/astro-sdk/pull/1385/commits/230add9e1d087f9fa6b1ae007e684aa8360db1a6) removed it

File: python-sdk/src/astro/sql/operators/cleanup.py [hunk 4]
```diff
@@ -232,9 +234,18 @@ def resolve_tables_from_tasks(  # noqa: C901
...
+                        for t in task.output.resolve(context):
+                            if isinstance(t, BaseTable):
+                                res.append(t)
+                    else:
+                        from airflow.models.xcom_arg import XComArg
+
+                        task_output = XComArg(operator=task)
+                        for t in task_output.resolve(context):
+                            if isinstance(t, BaseTable):
+                                res.append(t)
```
Comment (kaxil) [inline python-sdk/src/astro/sql/operators/cleanup.py:247]:
```suggestion
                try:
                    task_output = task.output
                except AttributeError:
                    task_output = XComArg(operator=task)
                
                for t in task_output.resolve(context):
                    if isinstance(t, BaseTable):
                        res.append(t)

```

can be simplified

**Review by kaxil — APPROVED** [review]

**Pull request merged**